services/api.service.ts

```diff
@@ -300,61 +335,103 @@ export class APIService {
+            })
+            .pipe(
+                catchError((err, caught) => this.handleError(err, caught))
+            );
     }
 
     public deleteRecord(id: number): Observable<Record> {
         return this.httpClient.delete(this.buildAbsoluteUrl('records/' + id))
-        .pipe(
-            catchError((err, caught) => this.handleError(err, caught))
-        );
+            .pipe(
+                catchError((err, caught) => this.handleError(err, caught))
+            );
     }
 
     public deleteRecords(datasetId: number, recordIds: number[]): Observable<void> {
         // httpClient.delete method doesn't accept a body argument, so use request as a work-around
-        return this.httpClient.request('DELETE', this.buildAbsoluteUrl('datasets/' + datasetId + '/records/'), {
-            body: recordIds
-        })
-        .pipe(
-            catchError((err, caught) => this.handleError(err, caught))
-        );
+        return this.httpClient.request('DELETE', this.buildAbsoluteUrl('datasets/' + datasetId + '/records/'),
+            {
+                body: recordIds
+            })
+            .pipe(
+                catchError((err, caught) => this.handleError(err, caught))
+            );
     }
 
     public deleteAllRecords(datasetId: number): Observable<void> {
         // httpClient.delete method doesn't accept a body argument, so use request as a work-around
-        return this.httpClient.request('DELETE', this.buildAbsoluteUrl('datasets/' + datasetId + '/records/'), {
-            body: JSON.stringify('all')
-        })
-        .pipe(
-            catchError((err, caught) => this.handleError(err, caught))
-        );
+        return this.httpClient.request('DELETE', this.buildAbsoluteUrl('datasets/' + datasetId + '/records/'),
+            {
+                body: JSON.stringify('all')
+            })
+            .pipe(
+                catchError((err, caught) => this.handleError(err, caught))
+            );
+    }
+
+    public getRecordMedia(recordId: number): Observable<Media[]> {
+        const params = {
+            record: recordId.toString()
+        };
+
+        return this.httpClient.get(this.buildAbsoluteUrl('media'), {
+                params: params
+            })
+            .pipe(
+                catchError((err, caught) => this.handleError(err, caught))
+            );
+    }
+
+    public uploadRecordMediaBinary(recordId: number, file: File): Observable<Media> {
+        const formData: FormData = new FormData();
+
+        formData.append('record', recordId.toString());
+        formData.append('file', file, file.name);
+
+        // the content-type will be inferred from formData
+        return this.httpClient.post(this.buildAbsoluteUrl('media'), formData)
+            .pipe(
+                catchError((err, caught) => this.handleError(err, caught))
+            );
+    }
+
+    public uploadRecordMediaBase64(recordId: number, file: string): Observable<Media> {
```

Comment:
Maybe renamed the parameter `file` to be `base64String` or something like that. I'm not sure now.